### PR TITLE
Refactor/report-filters

### PR DESCRIPTION
**Pull request by thebeyondr**
image

- adds retractable sidebar (becomes drawer on mobile)
- adds useFilter hook
- adds Filter Provider
- adds util function for filtering
- add util function for creating options
- moves voicedeck stats
- makes report cards a link
- updates report page layout

### TODO
- [x] Add back sorting
- [x] Add left padding to reports display
- [x] Add no filter results fallback
- [x] Fix report header on mobile

### LOOM
https://www.loom.com/share/682c4b6929e54c799e17925ed6d1e68a

### CODE REVIEW

File: components/reports/reports-view.tsx [hunk 1]
```diff
@@ -11,213 +10,172 @@ import {
...
+import {
+	createFilterOptions,
+	filterReports,
+	sortingOptions,
+} from "@/lib/search-filter-utils";
+import type { ISortingOption, Report } from "@/types";
 import Fuse from "fuse.js";
-import Link from "next/link";
 import { useMemo, useState } from "react";
+import { Button } from "../ui/button";
```
Comment (0xRowdy) [inline components/reports/reports-view.tsx:24]: use @ for imports
```suggestion
import { Button } from "@/components/ui/button";
```

File: components/reports/states-combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	CommandItem,
+	CommandList,
+} from "@/components/ui/command";
+import {
+	Popover,
+	PopoverContent,
+	PopoverTrigger,
+} from "@/components/ui/popover";
+import { cn } from "@/lib/utils";
+import { Badge } from "../ui/badge";
```
Comment (0xRowdy) [inline components/reports/states-combobox.tsx:21]:
```suggestion
import { Badge } from "@/components/ui/badge";
```

File: types/index.ts [hunk 3]
```diff
@@ -81,42 +81,42 @@ export interface Report {
...
+  contract: string;
+  // biome-ignore lint: type definition imported from @hypercerts-org/sdk
+  tokenID: any;
+  // biome-ignore lint: type definition imported from @hypercerts-org/sdk
+  creator?: any | null;
+  id: string;
+  // biome-ignore lint: type definition imported from @hypercerts-org/sdk
+  owner?: any | null;
+  // biome-ignore lint: type definition imported from @hypercerts-org/sdk
+  totalUnits?: any | null;
```
Comment (0xRowdy) [inline types/index.ts:94]: What up with all the any types?

**Review by 0xRowdy — APPROVED** [review]
lgtm

File: components/reports/reports-filters.tsx [hunk 4]
```diff
@@ -1,212 +1,228 @@
...
 							<DrawerClose>
 								<Button
 									className="px-24 py-4"
-									onClick={() => handleApplyFilters()}
 								>
 									Apply
 								</Button>
 							</DrawerClose>
-						</DrawerFooter>
+						</DrawerFooter> */}
```
Comment (CJ-Rose) [inline components/reports/reports-filters.tsx:203]: removal of "apply" button / separate handling of each filter selection causes mobile drawer to close on any click inside the open drawer. This allows the user to only apply a single filter each time the filter drawer is opened.

File: components/reports/reports-header.tsx [hunk 5]
```diff
@@ -1,184 +1,152 @@
...
+						activeSort ? sortingOptions[activeSort].label : "Sort by"
+					}
+				/>
+			</SelectTrigger>
+			<SelectContent>
+				<SelectGroup>
+					{Object.values(sortingOptions).map((option) => (
+						<SelectItem
+							key={option.value}
+							value={option.value}
```
[the remark this answers is not in the document]
Comment (thebeyondr) [reply components/reports/reports-header.tsx:71]: not that big a diff, I get to work with a tighter set of values I know are unique anyways

File: types/index.ts (hunk 3 above)
Comment (thebeyondr) [reply types/index.ts:94]: not sure, @baumstern, can you look into this?

File: components/reports/report-card.tsx [hunk 6]
```diff
@@ -33,38 +36,51 @@ const ReportCard: React.FC<ReportCardProps> = ({
...
+			<Card
+				key={hypercertId}
+				className="w-[250px] md:w-[280px] rounded-3xl bg-vd-beige-100 text-vd-blue-900 hover:scale-105 ease-[cubic-bezier(0.76, 0, 0.24, 1)] transition-transform duration-300 shadow-none hover:shadow-md"
+			>
+				<CardHeader className="p-0">
+					<img
+						src={image}
+						alt={title}
+						className="w-full h-44 overflow-clip object-cover object-center rounded-t-3xl"
+					/>
```
[the remark this answers is not in the document]
Comment (thebeyondr) [reply components/reports/report-card.tsx:49]: its always annoying to work with and provides little benefit imo

File: components/reports/reports-filters.tsx [hunk 7]
```diff
@@ -1,212 +1,228 @@
...
+			newFilter = [
+				...newFilter,
+				...Array.from(existingStates).map(
+					(stateValue) => ["state", stateValue] as [string, string],
+				),
+			];
+		}
+
+		updateSearchParams(newFilter);
+	};
```
[the remark this answers is not in the document]
Comment (thebeyondr) [reply components/reports/reports-filters.tsx:77]: thanks for pointing this out! the issue was how it was filtering in `search-filter-utils`

**Comment by CJ-Rose** [comment]
@thebeyondr i see a handful of ui styling issues that i haven't flagged here yet. Thinking better to let you get this merged to start QA asap, then I can take on style fixes and additional feedback from there?

**Comment by thebeyondr** [comment]
> @thebeyondr i see a handful of ui styling issues that i haven't flagged here yet. Thinking better to let you get this merged to start QA asap, then I can take on style fixes and additional feedback from there?

yes, thanks for understanding. I'm working on the Drawer bug but its fighting back. Should be able to gather everything in one branch after that

**Pull request merged**